src/services/ActivityService.ts

```diff
@@ -38,29 +39,68 @@ export class ActivityService {
             throw new HTTPResponse(400, { error });
         }
 
-        // Check if staff already has an ongoing activity
-        const ongoingCount: number = await this.dbClient.getOngoingByStaffId(activity.testerStaffId)
-            .then((result: DocumentClient.QueryOutput): number => {
-                return result.Count as number;
-            })
-            .catch((error: AWSError) => {
-                throw new HTTPResponse(error.statusCode, { error: `${error.code}: ${error.message} At: ${error.hostname} - ${error.region} Request id: ${error.requestId}` });
-            });
-
-        if (ongoingCount && ongoingCount > 0) {
-            throw new HTTPResponse(403, { error: HTTPRESPONSE.ONGOING_ACTIVITY_STAFF_ID + " " + activity.testerStaffId + " " + HTTPRESPONSE.ONGING_ACTIVITY});
+        // Check if parentId is not null if activityType is 'wait' or 'accountable time' and is null when activityType is 'visit'
```

Comment:
The createActivity function became very big, can you please split it in multiple functions?

Reply:
Refactored the function.